pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -1523,4 +1523,7 @@ self: super: {
     sha256 = "sha256:0xbfhzhzg94b4r5qy5dg1c40liswwpqarrc2chcwgfbfnrmwkfc2";
   });
 
+  # Depends on selective >= 0.4, but the default of selective is 0.3
+  headed-megaparsec = addBuildDepend super.headed-megaparsec super.selective_0_4_1;
```

Comment:
I think normally we write this like the following:

```nix
  headed-megaparsec = super.headed-megaparsec.override {
    selective = self.selective_0_4_1;
  };
```

Please fix this up (and make sure it still compiles correctly).